analyses/cell-type-ewings/03-copykat.Rmd

```diff
@@ -0,0 +1,402 @@
+
+# read in full gene by cell copy number detection results 
+full_ck_results_df <- full_ck_result_paths |> 
+  purrr::map(readr::read_tsv) |> 
+  dplyr::bind_rows(.id = "reference_used")
+```
+
+
+## CopyKAT results
+
+Below we look at the heatmaps produced by `CopyKAT`. 
+
+### Heatmap without reference
+
+![](`r file.path(results_dir, "no_normal", "SCPCL000822_copykat_heatmap.jpeg")`)
+
+### Heatmap with endothelial cells as reference
+
+![](`r file.path(results_dir, "endothelial_normal", "SCPCL000822_copykat_heatmap.jpeg")`)
+
+Both of these heatmaps appear to show some increase in observed CNVs in aneuploid vs. diploid which is to be expected. 
+
+### UMAP
+
+Below we prepare and plot a UMAP that shows which cells are classified as diploid, aneuploid, and not defined by `CopyKAT`. 
+We show a side by side UMAP with results from running `CopyKAT` both with and without a reference of normal cells. 
+
+```{r}
+umap_df <- sce |> 
+  scuttle::makePerCellDF(use.dimred = "UMAP") |> 
+  # replace UMAP.1 with UMAP1
+  dplyr::rename_with(
+        \(x) stringr::str_replace(x, "^UMAP\\.", "UMAP")
+      )
+
+cnv_df <- umap_df |> 
+  # first add manual annotations
+  dplyr::left_join(manual_classifications_df) |> 
+  # now add copykat results
+  dplyr::left_join(ck_results_df, by = c("barcodes" = "cell.names"))
+```
+
+```{r}
+ggplot(cnv_df, aes(x = UMAP1, y = UMAP2, color = copykat.pred)) +
+  geom_point(alpha = 0.5, size = 0.5) +
+  theme_bw() +
+  facet_wrap(vars(reference_used))
+```
+
+If we look at the UMAPs shown in `01-marker-gene-tumor-classification.Rmd`, the cells that are labeled aneuploid here are consistent with where we saw marker gene expession for tumor cells. 
+It also looks like there is not a visible difference between using a reference of normal cells and not using a reference. 
+
+### Validate common CNAs found in Ewing sarcoma
+
+To validate some of these annotations, we can also look at some commonly found copy number variations found in Ewing sarcoma patients. 
+There are a few known copy number variations in Ewing's sarcoma: 
+
+- Gain of Chr8
+- Gain of Chr12
+- Gain of Chr1p
+- Loss of Chr16q
+
+Although these are the most frequent, there are patients who do not have any of these alterations and patients that only have some of these alterations. 
+See [Tirode et al.,](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4264969/) and [Crompton et al.](https://doi.org/10.1158/2159-8290.CD-13-1037). 
+
+`CopyKAT` outputs a matrix that contains the estimated copy numbers for each gene in each cell. 
+We can read that in and look at the mean estimated copy numbers for each chromosome across each cell. 
+We might expect that tumor cells would show an increased estimated copy number in Chr8, Chr12, and/or Chr 1 and a loss of Chr16. 
+
+```{r}
+# for every cell, calculate the mean detection level across all genes in a given chromosome
+full_cnv_df <- full_ck_results_df |> 
+  tidyr::pivot_longer(cols = -c(
```

Comment:
You might want to remove the `NA` chromosome values somewhere in here. Not sure where they are coming from but they do add a row to your plot. (Maybe Y chromosome genes? I assume it is labelling X as 23)